bot/exts/moderation/infraction/_scheduler.py

```diff
@@ -337,10 +358,17 @@ async def deactivate_infraction(
             log.warning(f"Failed to deactivate infraction #{id_} ({type_}): bot lacks permissions.")
             log_text["Failure"] = "The bot lacks permissions to do this (role hierarchy?)"
             log_content = mod_role.mention
+        except discord.NotFound:
+            log.info(f"Can't pardon {infraction['type']} for user {infraction['user']} because user left from guild.")
         except discord.HTTPException as e:
-            log.exception(f"Failed to deactivate infraction #{id_} ({type_})")
-            log_text["Failure"] = f"HTTPException with status {e.status} and code {e.code}."
-            log_content = mod_role.mention
+            if e.code == 10007:
+                log.info(
+                    f"Can't pardon {infraction['type']} for user {infraction['user']} because user left from guild."
+                )
+            else:
+                log.exception(f"Failed to deactivate infraction #{id_} ({type_})")
+                log_text["Failure"] = f"HTTPException with status {e.status} and code {e.code}."
+                log_content = mod_role.mention
```

Comment:
A similar thing needs to be done (but with a different message) when the user isn't found, since that is still considered a failure.